v1/ao/internal/reporter/oboe_test.go

```diff
@@ -72,7 +72,7 @@ func TestTokenBucket(t *testing.T) {
 			perConsumerRate := newTokenBucket(15, 1)
 			for i := 0; i < iters; i++ {
 				sampled := perConsumerRate.consume(1)
-				ok := b.count(sampled, i%2 == 0, true)
+				ok := b.count(sampled, false, true)
```

Comment:
Are we going to have a test case to verify that if `hasMetadata` is `true` and `sampled` is false, then we will NOT increment the `ThroughTraceCount` (assuming this is the goal of this fix :) )